File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot extends Parrot {

    private static final double LOAD_FACTOR = 9.0;
    private final int numberOfCoconuts;

    public AfricanParrot(int numberOfCoconuts) {
        super();
        this.numberOfCoconuts = numberOfCoconuts;
    }

    @Override
    public double getSpeed() {
        return Math.max(MINIMUM_SPEED, BASE_SPEED - LOAD_FACTOR * numberOfCoconuts);
    }

}

File: Kotlin/src/main/kotlin/parrot/AfricanParrot.kt
package parrot

import kotlin.math.max

class AfricanParrot(private val numberOfCoconuts: Int) : Parrot() {

    val speed: Double = max(MINIMUM_SPEED, BASE_SPEED - LOAD_FACTOR * numberOfCoconuts)

    companion object {
        const val LOAD_FACTOR: Double = 9.0
    }
}
